test/model/binding.dart

```diff
@@ -508,6 +509,20 @@ typedef FirebaseMessagingRequestPermissionCall = ({
 });
 
 class FakeAndroidNotificationHostApi implements AndroidNotificationHostApi {
+  // TODO(?): Find a better way to handle this. This member is exported from
+  //   the Pigeon generated class but are not used for this fake class,
+  //   so return the default value.
+  @override
+  // ignore: non_constant_identifier_names
+  final BinaryMessenger? pigeonVar_binaryMessenger = null;
+
+  // TODO(?): Find a better way to handle this. This member is exported from
+  //   the Pigeon generated class but are not used for this fake class,
+  //   so return the default value.
+  @override
+  // ignore: non_constant_identifier_names
+  final String pigeonVar_messageChannelSuffix = '';
```

Comment:
I wonder if `dartHostTestHandler` helps, but it is also subject to removal (https://github.com/flutter/flutter/issues/159886). Mocking `HostApi` this way seems fine to me. Maybe these `pigeonVar` prefixed variables should be private in the generated code.